src/utils/spark.py

```diff
@@ -71,6 +87,21 @@ def __init__(self, app_name: str, password_file_path: str) -> None:
 
         self.spark = SparkSession.builder.appName(self.app_name).getOrCreate()
 
+    def run_gh_workflow(self):
```

Comment:
**[Thought, non-blocking]** Two thoughts on the design of this method, neither of which are particularly urgent:

1. Does it actually make sense for the `SharedSparkSession` object to own this logic? Up until now, `SharedSparkSession` has owned responsibility managing our connection to the Spark cluster; this change also gives it responsibility for managing our connection to GitHub, which feels to me like a distinct concern. Two ways we could think about separating these concerns:  
    1. Make this a bare, non-class function that takes an app ID, PEM path, repo, and workflow as arguments; keep things simple and function-oriented unless/until we decide we need to do more things with a GitHub connection such that it makes sense to factor out the connection management
    2. Match the design of `SharedSparkSession` and `SparkJob` by designing `SharedGitHubSession` and `GitHubJob` classes to split responsibility for connection/job management
2. The name `run_gh_workflow` doesn't actually make it clear _which_ workflow will run, and I could imagine us wanting to trigger multiple workflows in the future. Two options for clarity and future-proofing:
    1. Call the method something more specific like `run_gh_workflow_tests`
    2. Factor out the repo and workflow names into arguments, such that it can be called like `session.run_gh_workflow("data-architecture", "test-dbt-models")`

I don't feel super strongly about any of this, and ultimately I would be fine with us going with the current design and keeping an eye out for smells that we need more general abstractions. Up to you!

Reply:
I am dumb and should've just gone with option 2. Reworked in 9662859!